diff --git a/arch/riscv/boot/dts/allwinner/sun20i-d1-mangopi-mq-pro.dts b/arch/riscv/boot/dts/allwinner/sun20i-d1-mangopi-mq-pro.dts
index e2bb6bc16c13..fc84c04f334c 100644
--- a/arch/riscv/boot/dts/allwinner/sun20i-d1-mangopi-mq-pro.dts
+++ b/arch/riscv/boot/dts/allwinner/sun20i-d1-mangopi-mq-pro.dts
@@ -16,12 +16,20 @@ / {
 	aliases {
 		ethernet0 = &rtl8723ds;
 		serial0 = &uart0;
+		serial4 = &uart4;
 	};
 
 	chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
+	/* Kernel panic otherwise */
+	/* OF: reserved mem: Failed to allocate memory for reserved_mem array with err: -12 */
+	memory@40000000 {
+		device_type = "memory";
+		reg = <0x40000000 0x20000000>; /* start 0x40000000, size 512MB */
+	};
+
 	leds {
 		compatible = "gpio-leds";
 
@@ -109,6 +117,11 @@ &ohci1 {
 
 &pio {
 	vcc-pe-supply = <&reg_avdd2v8>;
+
+	uart4_pb2_pins: uart4-pb2-pins {
+		pins = "PB2", "PB3";
+		function = "uart4";
+    };
 };
 
 &uart0 {
@@ -131,6 +144,12 @@ bluetooth {
 	};
 };
 
+&uart4 {
+	pinctrl-0 = <&uart4_pb2_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
 &usb_otg {
 	dr_mode = "peripheral";
 	status = "okay";
